src/converters/toAfmResultSpec.ts

```
@@ -1,415 +1,229 @@
-        const pieOnlyMeasures = visObj.type === PIE && categories.length === 0;
-        const stackedChart = isChartStackedOrSegmented(visObj);
-
-        const measureGroupIndex = pieOnlyMeasures || stackedChart || visObj.type === TABLE ? 1 : 0;
-        itemIdentifiersList[measureGroupIndex].push('measureGroup');
-    }
-
-    return itemIdentifiersList.map(itemIdentifiers => generateDimension(itemIdentifiers, visObj, afm));
+    return nativeTotals.map(total => ({
+        measureIdentifier: total.measureIdentifier,
+        attributeIdentifiers: []
+    }));
 }
 
-function shouldConvertTotals(itemIdentifiers: AFM.Identifier[], visObj: VisObj.IVisualizationObjectContent): boolean {
-    if (!isEmpty(itemIdentifiers)) {
-        const hasTotals = !isEmpty(visObj.buckets.totals);
-        const containsMeasureGroup = itemIdentifiers.some(item => item === 'measureGroup');
-        return hasTotals && !containsMeasureGroup;
-    }
+function getAttributes(buckets: VisualizationObject.IBucket[]): VisualizationObject.IVisualizationAttribute[] {
+    return buckets.reduce(
+        (result: VisualizationObject.IVisualizationAttribute[], bucket: VisualizationObject.IBucket) => {
+            const items =
+                filter(bucket.items, VisualizationObject.isAttribute) as VisualizationObject.IVisualizationAttribute[];
 
-    return false;
+            return result.concat(items);
+    }, []);
 }
 
-function convertTotals(itemIdentifiers: AFM.Identifier[], visObj: VisObj.IVisualizationObjectContent,
-                       afm: AFM.IAfm): AFM.ITotalItem[] {
-    if (!shouldConvertTotals(itemIdentifiers, visObj)) {
-        return [];
-    }
+function convertSorting(visObj: VisualizationObject.IVisualizationObjectContent): AFM.SortItem[] {
+    if (visObj.properties) {
+        let properties = {};
+        try {
+            properties = JSON.parse(visObj.properties);
+        } catch (e) {
+            // properties is a malformed json
```

What I know this is in general bad practice to swallow exceptions... there should be another way how to handle it...
I can imagine that user could be confused if he puts invalid JSON to properties but he gets nothing as a feedback that something is wrong...

at least there could be console.warning or error which is less bad option then just pretending that nothing bad happen.